### ISSUE TITLE
Error when trying to 'pachctl delete-all'

### ISSUE THREAD

**Issue by cfga** [pachyderm/pachyderm#3678]
**What happened?**:

An error is thrown when trying to delete all pipelines/repos. It happens when a pipeline output repo was created upfront and the pipelines are created in reverse order with stats enabled. The error:

`delete branch repo:<name:"pipeline-1" > name:"master" : branch master has [repo:<name:"pipeline-2" > name:"master"  repo:<name:"pipeline-2" > name:"stats" ] as subvenance, deleting it would break those branches`

**Environment**:

Minikube/VirtualBox Version 5.2.18 r124319 (Qt5.6.3)

Kubernetes:
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-26T00:05:06Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-25T15:45:25Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```
Pachyderm version:

```
pachctl             1.8.6
pachd               1.8.6
```

**How to reproduce:**

This is a trivial example of two pipelines, one feeding its output into another, with the `pipeline-1` output repo created before any of the pipelines:

```python
import os

import python_pachyderm.client.pps.pps_pb2 as proto
from python_pachyderm import PfsClient, PpsClient

PACHD_SERVICE_HOST = os.getenv("PACHD_SERVICE_HOST", "localhost")
PACHD_SERVICE_PORT_API_GRPC_PORT = os.getenv("PACHD_SERVICE_PORT_API_GRPC_PORT", 30650)

pfs_client = PfsClient(PACHD_SERVICE_HOST, PACHD_SERVICE_PORT_API_GRPC_PORT)
pps_client = PpsClient(PACHD_SERVICE_HOST, PACHD_SERVICE_PORT_API_GRPC_PORT)

if __name__ == "__main__":
    pfs_client.create_repo('input')
    pfs_client.create_repo('pipeline-1')

    pps_client.create_pipeline(
        "pipeline-2",
        transform=proto.Transform(cmd=["/bin/bash"],
                                  stdin=[
                                      "cat /pfs/input/myfile > /pfs/out/myfile"
                                  ],
                                  image="ubuntu:18.04"),
        parallelism_spec=proto.ParallelismSpec(constant=1),
        output_branch='master',
        input=proto.Input(cross=[
            proto.Input(pfs=proto.PFSInput(glob="/*", repo="pipeline-1", branch='master', name='input'))
        ]),
        enable_stats=True
    )

    pps_client.create_pipeline(
        "pipeline-1",
        transform=proto.Transform(cmd=["/bin/bash"],
                                  stdin=[
                                    "cat /pfs/input/myfile > /pfs/out/myfile"
                                  ],
                                  image="ubuntu:18.04"),
        parallelism_spec=proto.ParallelismSpec(constant=1),
        output_branch='master',
        input=proto.Input(cross=[
            proto.Input(pfs=proto.PFSInput(glob="/*", repo="input", branch='master', name='input'))
        ]),
        enable_stats=True
    )

    with pfs_client.commit('input', 'master') as c:
        pfs_client.put_file_bytes(c, 'myfile', b'PPPPP')

```

Then the aforementioned error is thrown when I either say `pachctl delete-all` _or_ call these methods in this order:

```python
pps_client.delete_all()
pfs_client.delete_all()
```

Calling the methods in the following order (pfs first) results in a successful deletion:

```python
pfs_client.delete_all()
pps_client.delete_all()
```

**Comment by ysimonson** [comment]
Possibly related to #3225 - are you on 1.8.7 or 1.9.0?

**Comment by JoeyZwicker** [comment]
it looks like he's on 1.8.6

**Comment by JoeyZwicker** [comment]
@adelelopez Is this something that is related or may be fixed by your current subvenance changes? Delete-all _should_ be able to force it's way through these kinds of situations regardless, but of course we want to understand how things got into this state.

**Comment by Tryneus** [comment]
@JoeyZwicker - I believe this was fixed in #3225, but @cfga is using an earlier version.

**Comment by cfga** [comment]
@Tryneus @JoeyZwicker @ysimonson checked with 1.9.0rc1 – it works. But for now we decided to use 1.8.6 (with a workaround, apparently) due to some issues with accessing v1.9.0rc1 through python client and because it is still a release candidate. But it is unrelated, so I close this issue. Thank you for the help!
